### ISSUE TITLE
BUG: Pandas rolling removes imaginary part of complex 

### ISSUE THREAD

**Issue by fdelaigue** [pandas-dev/pandas#46619]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
pd.DataFrame([1j,1+2j]).rolling(2).apply(lambda x: print(x) is None)

Output:
0   0
1   1
dtype: float64
	0
0	NaN
1	1
```

### Issue Description

.rolling() casts complex128 values into float64, discarding the imaginary part.

### Expected Behavior

.rolling() passes complex128 values as is without removing the imaginary part.

Expected Output:
```python
import pandas as pd
pd.DataFrame([1j,1+2j]).rolling(2).apply(lambda x: print(x) is None)

0   0+1j
1   1+2j
dtype: complex128
	0
0	NaN
1	1
```
### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.10.0.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.2.0
Version          : Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.4.1
numpy            : 1.21.5
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 60.2.0
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.8.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 8.1.1
pandas_datareader: 0.10.0
bs4              : 4.10.0
bottleneck       : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.1
numba            : 0.55.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

**Comment by simonjayhawkins** [comment]
Thanks @fdelaigue for the report

> .rolling() casts complex128 values into float64, discarding the imaginary part.

did you not get a warning?

```
/home/simon/miniconda3/envs/pandas-1.4.2/lib/python3.10/site-packages/pandas/core/window/rolling.py:348: ComplexWarning: Casting complex values to real discards the imaginary part
  values = ensure_float64(values)
```

relabeling as an enhancement request and not a bug report.

**Comment by fdelaigue** [comment]
no warning. just the result. 

additionally, adding astype(np.complex64) still returns float64: 
`pd.DataFrame([1j,1+2j]).astype(np.complex64).rolling(2).apply(lambda x: print(x) is None)`
Output:
`<div class="output_container" style="width: 555px; color: rgb(212, 212, 212); font-family: -apple-system, &quot;system-ui&quot;, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; position: absolute; overflow: hidden; max-height: 82px; top: 74px; height: 82px;"><div id="a788467f-35f6-4165-94eb-e1136db320de" class="output" output-mime-type="application/vnd.code.notebook.stdout" style="font-size: var(--notebook-cell-output-font-size); width: var(--notebook-output-width); margin-left: var(--notebook-output-left-margin); padding: 8px; box-sizing: border-box; border-right-color: ; border-right-style: ; border-right-width: ; border-bottom-color: ; border-bottom-style: ; border-bottom-width: ; border-left-color: ; border-left-style: ; border-left-width: ; border-image-source: ; border-image-slice: ; border-image-width: ; border-image-outset: ; border-image-repeat: ; background-color: var(--theme-notebook-output-background); border-top: none !important; overflow-x: auto; position: absolute; top: 0px; left: 0px;"><span class="output-stream" style="line-height: 22px; font-family: var(--notebook-cell-output-font-family); white-space: pre-wrap; overflow-wrap: break-word; font-size: var(--notebook-cell-output-font-size); user-select: text; cursor: auto;"><span><span class=""><span>0   0
</span><span>1   1
</span><span>dtype: float64
</span></span></span></span></div></div><div class="output_container" style="width: 555px; color: rgb(212, 212, 212); font-family: -apple-system, &quot;system-ui&quot;, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; position: absolute; overflow: hidden; max-height: 82px; top: 156px; height: 82px;"><div id="a5f65368-7e63-489d-862f-9a51d250ed52" class="output" style="font-size: var(--notebook-cell-output-font-size); width: var(--notebook-output-width); margin-left: var(--notebook-output-left-margin); padding: 8px; box-sizing: border-box; border-right-color: ; border-right-style: ; border-right-width: ; border-bottom-color: ; border-bottom-style: ; border-bottom-width: ; border-left-color: ; border-left-style: ; border-left-width: ; border-image-source: ; border-image-slice: ; border-image-width: ; border-image-outset: ; border-image-repeat: ; background-color: var(--theme-notebook-output-background); border-top: none !important; overflow-x: auto; position: absolute; top: 0px; left: 0px;"><div><div>

  | 0
-- | --
NaN
1

</div></div></div></div>`

**Comment by lithomas1** [comment]
This isssue is still not addressed, but I will close and move to #53214, since that one is more general.
